modules/oe_translation_cdt/modules/oe_translation_cdt_mock/src/Plugin/ServiceMock/StatusApi.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\oe_translation_cdt_mock\Plugin\ServiceMock;
+
+use Drupal\Core\Site\Settings;
+use Drupal\oe_translation_cdt\Mapper\LanguageCodeMapper;
+use Drupal\oe_translation_remote\TranslationRequestRemoteInterface;
+use GuzzleHttp\Psr7\Response;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Intercepts status request made to CDT.
+ *
+ * @ServiceMock(
+ *   id = "oe_translation_cdt_status_api",
+ *   label = @Translation("CDT mocked status responses for testing."),
+ *   weight = 0,
+ * )
+ */
+class StatusApi extends ServiceMockBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEndpointUrl(): string {
+    return Settings::get('cdt.status_api_endpoint');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getEndpointResponse(RequestInterface $request): ResponseInterface {
+    // The entity uses the same identifier as the mocked CDT request.
+    $parameters = $this->getRequestParameters($request);
```

Comment:
I decided not to keep any additional data structures that would be used for the mock.
This mock fetches the translation request entity and constructs a matching JSON response.
This response should not actually update the request data.

To fake the translation process moving forward, we will use the fake callbacks.